### ISSUE TITLE
Lint/MixedRegexpCaptureTypes fails for multiline regex with comments

### ISSUE THREAD

**Issue by bkuhlmann** [rubocop-hq/rubocop#8845]
## Expected behavior

I would expect Rubocop to not flag multi-line regular expressions with comments as invalid.

## Actual behavior

End up with the following error:

    Lint/MixedRegexpCaptureTypes: Do not mix named captures and numbered captures in a Regexp literal.

## Steps to reproduce the problem

You can recreate the issue as follows:

``` ruby
# frozen_string_literal: true

SUCCESS = /
  (?<key>\A.+)
/x.freeze

FAILURE = /
  (?<key>\A.+)  # Key (anchored to start of line).
/x.freeze
```

If you save the above to `example.rb` and run `rubocop example.rb`, you'll get the following error:

``` 
Inspecting 1 file
W

Offenses:

example.rb:3:19: W: Lint/MixedRegexpCaptureTypes: Do not mix named captures and numbered captures in a Regexp literal.
PATTERN_FAILURE = / ...
                  ^

1 file inspected, 1 offense detected, 1 offense auto-correctable
```

## RuboCop version

```
0.92.0 (using Parser 2.7.1.5, rubocop-ast 0.7.1, running on ruby 2.7.2 x86_64-darwin19)
```

**Comment by dvandersluis** [comment]
I think this was fixed by #8803 already!

**Comment by bkuhlmann** [comment]
:bow: @dvandersluis Perfect. I was searching for open/closed issues related to `Lint/MixedRegexpCaptureType` but didn't see that at all. I look forward to the next release.
